integration-cli/docker_cli_wait_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"os/exec"
+	"testing"
+)
+
+// non-blocking wait with 0 exit code
+func TestWaitNonBlockedExitZero(t *testing.T) {
+	defer deleteAllContainers()
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "sh", "-c", "true")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+	containerID := stripTrailingCharacters(out)
+
+	status := "true"
+	for status != "false" {
+		runCmd = exec.Command(dockerBinary, "inspect", "--format='{{.State.Running}}'", containerID)
+		status, _, err = runCommandWithOutput(runCmd)
+		if err != nil {
+			t.Fatal(status, err)
+		}
```

Comment:
and to prevent it from hanging for 45 minutes, like its doing now, you may want to:
1 - add a sleep after each to pause a second
2 - stop after, say, 60 tries/seconds and fail because something went seriously wrong then